src/assets.rs

```diff
@@ -280,7 +293,11 @@ impl HighlightingAssets {
     fn get_first_line_syntax(&self, reader: &mut InputReader) -> Option<&SyntaxReference> {
         String::from_utf8(reader.first_line.clone())
             .ok()
-            .and_then(|l| self.get_syntax_set().find_syntax_by_first_line(&l))
+            .and_then(|l| {
+                self.get_syntax_set()
+                    .ok()
```

Comment:
This will conflate a missing syntax_set with the non-availability of a first line syntax. Maybe it's better to use an explicit `unwrap` here?

Reply:
Similar for the other `.ok()` calls.

Reply:
Hmm, wouldn't it be better to change to Result<Option> instead? So that we have three cases:
* Failure to look up
* Nothing was found
* Something was found

Reply:
It depends whether or not the `get_syntax_set` method can fail at this particular place. Based on the `.unwrap()` calls in similar methods below, I thought it could not fail here. In that case I would have preferred `.unwrap()` or `.expect()`. If it can indeed fail, `Result<Option<_>>` is definitely the way to go. You will probably need one of my [favorite helper functions](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.transpose)

Reply:
Aha, you probably missed that my `.unwrap()`s are in `mod tests { ...`, where I think it is (more) OK to take shortcuts. (And in `PrettyPrinter`, but there is can never fail, as motivated in the added comment.)

I decided to go full robustness, and that it was not enough to handle `~/.cache/bat/syntaxes.bin`. I also want to robustly (meaning: giving helpful message to the user, and not panicing) handle `echo garbage >! ~/.cache/bat/syntaxes.bin`. So my proposed quickfix in the lazy-load PR was not enough.

(I'm not going to bother protecting against flipped bits in the bat binary itself though...)

Anyway, I'll go through which code paths with `Option` that can never fail (because of them never being the first call to get_syntax_set()), and adjust accordingly. Thanks for the code review!

Reply:
I've cascaded the return of `Result` now to all necessary places, and all regression tests pass, so would be great if you could take a second look.

Reply:
> Aha, you probably missed that my `.unwrap()`s are in `mod tests { ...`? I think it is (more) OK to take shortcuts in tests. (I also do .unwrap() in `PrettyPrinter`, but there is can never fail, as motivated in the added comment.)

I saw the unwrap in PrettyPrinter and thought that it might apply to these other situations. Thanks for clarifying.